ui/shared/EnsResolver.qml

```diff
@@ -0,0 +1,50 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Layouts 1.13
+import QtGraphicalEffects 1.13
+import "../imports"
+
+Item {
+    id: root
+    property bool isPending: false
+    readonly property string uuid: Utils.uuid()
+    readonly property var validateAsync: Backpressure.debounce(inpAddress, 600, function (inputValue) {
+        root.isPending = true
+        var name = inputValue.startsWith("@") ? inputValue.substring(1) : inputValue
+        walletModel.resolveENS(name, uuid)
+    });
+    signal resolved(string resolvedAddress)
+
+    function resolveEns(name) {
+        if (Utils.isValidEns(name)) {
+            root.validateAsync(name)
+        }
+    }
+    width: 12
+    height: 12
+
+    Loader {
+        anchors.fill: parent
+        sourceComponent: loadingIndicator
+        active: root.isPending
+    }
+
+    Component {
+        id: loadingIndicator
+        LoadingImage {
+            width: root.width
+            height: root.height
+        }
+    }
+
+    Connections {
+        target: walletModel
+        onEnsWasResolved: {
+            if (uuid !== root.uuid) {
```

Comment:
Same uuid question as the other PR since it was moved here haha
https://github.com/status-im/nim-status-client/pull/1257#discussion_r514286574

Reply:
So i guess when it comes to abstracting this out in to a component that can be placed anywhere, the idea was to only react to the request in the originating component. However, you've raised a really good point in that generally speaking, instead of UUID, we could use a combination of request parameters. The resulting difference would be that every component with the same request parameters would react to the response, where this may not be intended. With a UUID, only the component that originated the request would respond, which I feel is generally more the default assumption. Personally, I think we maybe should leave this for now, and if we have the intention to require this type of functionality later, we can alter the `EnsResolver` component to respond uniquely or universally. But definitely a great point!